internal/descriptor/descriptor.go

```diff
@@ -21,6 +21,20 @@ import (
 	"oras.land/oras/internal/docker"
 )
 
+// IsManifest checks if a descriptor describes a manifest.
+func IsManifest(desc ocispec.Descriptor) bool {
+	switch desc.MediaType {
+	case docker.MediaTypeManifest,
+		docker.MediaTypeManifestList,
+		ocispec.MediaTypeImageManifest,
+		ocispec.MediaTypeImageIndex,
+		"application/vnd.oci.artifact.manifest.v1+json":
```

Comment:
Should add docker manifest list in but remove `application/vnd.oci.artifact.manifest.v1+json`.
@Wwwsylvia What do you think? Also any chance to promote `IsManifest` and `IsImageManifest` to oras-go?

Reply:
I copied this from oras-go. It's an internal oras-go function.